nestedtensor/csrc/nested_node.cpp

```diff
@@ -9,27 +9,38 @@ namespace nested_tensor {
 using namespace torch::jit;
 using namespace torch::autograd::utils;
 
-std::string _NestedNode___str__(const TensorNode& nested_node) {
+void pretty_print_tensor(std::stringstream& result, const std::string& python_tensor_string, const std::string& tabs) {
+  auto tensor = strdup(python_tensor_string.c_str());
```

Comment:
I don't think you need to do this. You can use the copy constructor of std::string.

Reply:
strtok expects a char* so that doesn't work:(